contracts/staking/test/protocol_fees.ts

```diff
@@ -0,0 +1,361 @@
+            );
+            const tx = protocolFees.payProtocolFee.sendTransactionAsync(
+                makerAddress,
+                payerAddress,
+                constants.ZERO_AMOUNT,
+                { from: exchange, value: DEFAULT_PROTOCOL_FEE_PAID },
+            );
+            return expect(tx).to.revertWith(expectedError);
+        });
+
+        it('should revert if `protocolFeePaid` is different than the provided message value', async () => {
+            const differentProtocolFeePaid = DEFAULT_PROTOCOL_FEE_PAID.minus(50);
+            const expectedError = new StakingRevertErrors.InvalidProtocolFeePaymentError(
+                differentProtocolFeePaid,
+                DEFAULT_PROTOCOL_FEE_PAID,
+            );
+            const tx = protocolFees.payProtocolFee.sendTransactionAsync(
+                makerAddress,
+                payerAddress,
+                differentProtocolFeePaid,
+                { from: exchange, value: DEFAULT_PROTOCOL_FEE_PAID },
+            );
+            return expect(tx).to.revertWith(expectedError);
+        });
+
+        it('should call `transferFrom` in the proxy if no value is sent and the maker is not in a pool', async () => {
+            const receipt = await protocolFees.payProtocolFee.awaitTransactionSuccessAsync(
+                payerAddress, // This is an unregistered maker address
+                payerAddress,
+                DEFAULT_PROTOCOL_FEE_PAID,
+                { from: exchange, value: 0 },
+            );
+
+            // Ensure that the correct number of logs were recorded.
+            expect(receipt.logs.length).to.be.eq(1);
+
+            // Ensure that the correct log was recorded.
+            const log = logDecoder.decodeLogOrThrow(receipt.logs[0]) as LogWithDecodedArgs<
+                TestProtocolFeesERC20ProxyTransferFromCalledEventArgs
+            >;
+            expect(log.event).to.be.eq('TransferFromCalled');
+            expect(log.args.assetData).to.be.eq(WETH_ASSET_DATA);
+            expect(log.args.amount).bignumber.to.be.eq(DEFAULT_PROTOCOL_FEE_PAID);
+            expect(log.args.from).to.be.eq(payerAddress);
+            expect(log.args.to).to.be.eq(protocolFees.address);
+
+            // Verify that the end state is correct.
+            await verifyEndStateAsync(constants.NULL_BYTES32, constants.ZERO_AMOUNT);
+        });
+
+        it('should call `transferFrom` in the proxy and update `protocolFeesThisEpochByPool` if no value is sent and the maker is in a pool', async () => {
+            const receipt = await protocolFees.payProtocolFee.awaitTransactionSuccessAsync(
+                makerAddress,
+                payerAddress,
+                DEFAULT_PROTOCOL_FEE_PAID,
+                { from: exchange, value: 0 },
+            );
+
+            // Ensure that the correct number of logs were recorded.
+            expect(receipt.logs.length).to.be.eq(1);
+
+            // Ensure that the correct log was recorded.
```

Comment:
Wondering if we could move this into an actor or assertion wrapper, to follow similar patterns used elsewhere in our tests. Would also reduce a lot of the code that's repeated in these tests.

Reply:
Yeah I can work on that. It should be a very simple actor for now.